runtime/build.gradle.kts

```diff
@@ -2,15 +2,25 @@ import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget
 
 plugins {
     kotlin("multiplatform")
+    id("com.android.library")
     `maven-publish`
 }
 
+repositories {
+    google()
+}
+
 kotlin {
 
-    jvm {
-        withJava()
+    android {
+        publishLibraryVariants("release")
```

Comment:
Weirdly, the publishing of `android` artifacts seems to work differently than the other artifacts. 😢 
This also causes duplicate code to generate the artifact name.

Reply:
Oh, I think this is because the `publishing {}` block at the bottom of this file needs to be wrapped inside of `afterEvaluate {}` to work correctly for Android. See https://developer.android.com/studio/build/maven-publish-plugin for more details.

Reply:
I've tried your suggestion, but that didn't work. Which is expected as per the documentation here:
https://kotlinlang.org/docs/mpp-publish-lib.html#publish-an-android-library

I think this is because Android has variants by default and KMP wouldn't know what variants to publish unless some configuration tells it.

Reply:
I misunderstood which line of the code your original comment was referring to. The `publishLibraryVariants("release")` is indeed required. I meant that you can remove this block:

```kt
        mavenPublication {
            this.artifactId = "pbandk-$artifactId"
        }
```

if you wrap the `publishing {}` block in `afterEvaluate`.

Reply:
Yeah I tried that as well. IIRC: the Android artifact doesn't have an android suffix in its name.

So we could follow your approach, but then we would have to rename the Android artifact based on some metadata in the `publishing` block.

Reply:
Are you sure? I tried `./gradlew publishToMavenLocal` with your PR, and then I tried the same command with my changes applied on top of your PR, and both commands generated the same set of files (I made sure to delete my local maven repo in-between).

Here's the change I made:

```diff
diff --git a/runtime/build.gradle.kts b/runtime/build.gradle.kts
index 1aacbec..8a8788f 100644
--- a/runtime/build.gradle.kts
+++ b/runtime/build.gradle.kts
@@ -14,9 +14,6 @@ kotlin {
 
     android {
         publishLibraryVariants("release")
-        mavenPublication {
-            this.artifactId = "pbandk-$artifactId"
-        }
     }
 
     jvm()
@@ -170,13 +167,15 @@ tasks {
     }
 }
 
-publishing {
-    publications.withType<MavenPublication>().configureEach {
-        artifactId = "pbandk-$artifactId"
-        description = "Library for pbandk runtime protobuf code"
-        pom {
-            configureForPbandk()
+afterEvaluate {
+    publishing {
+        publications.withType<MavenPublication>().configureEach {
+            artifactId = "pbandk-$artifactId"
+            description = "Library for pbandk runtime protobuf code"
+            pom {
+                configureForPbandk()
+            }
+            addBintrayRepository(project, this)
         }
-        addBintrayRepository(project, this)
     }
 }
```

and here's the resulting set of published files:

```
pbandk-runtime
pbandk-runtime/0.10.0-SNAPSHOT
pbandk-runtime/0.10.0-SNAPSHOT/maven-metadata-local.xml
pbandk-runtime/0.10.0-SNAPSHOT/pbandk-runtime-0.10.0-SNAPSHOT.jar
pbandk-runtime/0.10.0-SNAPSHOT/pbandk-runtime-0.10.0-SNAPSHOT.module
pbandk-runtime/0.10.0-SNAPSHOT/pbandk-runtime-0.10.0-SNAPSHOT.pom
pbandk-runtime/maven-metadata-local.xml
pbandk-runtime-android
pbandk-runtime-android/0.10.0-SNAPSHOT
pbandk-runtime-android/0.10.0-SNAPSHOT/maven-metadata-local.xml
pbandk-runtime-android/0.10.0-SNAPSHOT/pbandk-runtime-android-0.10.0-SNAPSHOT-sources.jar
pbandk-runtime-android/0.10.0-SNAPSHOT/pbandk-runtime-android-0.10.0-SNAPSHOT.aar
pbandk-runtime-android/0.10.0-SNAPSHOT/pbandk-runtime-android-0.10.0-SNAPSHOT.module
pbandk-runtime-android/0.10.0-SNAPSHOT/pbandk-runtime-android-0.10.0-SNAPSHOT.pom
pbandk-runtime-android/maven-metadata-local.xml
pbandk-runtime-iosarm64
pbandk-runtime-iosarm64/0.10.0-SNAPSHOT
pbandk-runtime-iosarm64/0.10.0-SNAPSHOT/maven-metadata-local.xml
pbandk-runtime-iosarm64/0.10.0-SNAPSHOT/pbandk-runtime-iosarm64-0.10.0-SNAPSHOT-sources.jar
pbandk-runtime-iosarm64/0.10.0-SNAPSHOT/pbandk-runtime-iosarm64-0.10.0-SNAPSHOT.klib
pbandk-runtime-iosarm64/0.10.0-SNAPSHOT/pbandk-runtime-iosarm64-0.10.0-SNAPSHOT.module
pbandk-runtime-iosarm64/0.10.0-SNAPSHOT/pbandk-runtime-iosarm64-0.10.0-SNAPSHOT.pom
pbandk-runtime-iosarm64/maven-metadata-local.xml
pbandk-runtime-iosx64
pbandk-runtime-iosx64/0.10.0-SNAPSHOT
pbandk-runtime-iosx64/0.10.0-SNAPSHOT/maven-metadata-local.xml
pbandk-runtime-iosx64/0.10.0-SNAPSHOT/pbandk-runtime-iosx64-0.10.0-SNAPSHOT-sources.jar
pbandk-runtime-iosx64/0.10.0-SNAPSHOT/pbandk-runtime-iosx64-0.10.0-SNAPSHOT.klib
pbandk-runtime-iosx64/0.10.0-SNAPSHOT/pbandk-runtime-iosx64-0.10.0-SNAPSHOT.module
pbandk-runtime-iosx64/0.10.0-SNAPSHOT/pbandk-runtime-iosx64-0.10.0-SNAPSHOT.pom
pbandk-runtime-iosx64/maven-metadata-local.xml
pbandk-runtime-js
pbandk-runtime-js/0.10.0-SNAPSHOT
pbandk-runtime-js/0.10.0-SNAPSHOT/maven-metadata-local.xml
pbandk-runtime-js/0.10.0-SNAPSHOT/pbandk-runtime-js-0.10.0-SNAPSHOT-sources.jar
pbandk-runtime-js/0.10.0-SNAPSHOT/pbandk-runtime-js-0.10.0-SNAPSHOT.jar
pbandk-runtime-js/0.10.0-SNAPSHOT/pbandk-runtime-js-0.10.0-SNAPSHOT.module
pbandk-runtime-js/0.10.0-SNAPSHOT/pbandk-runtime-js-0.10.0-SNAPSHOT.pom
pbandk-runtime-js/maven-metadata-local.xml
pbandk-runtime-jvm
pbandk-runtime-jvm/0.10.0-SNAPSHOT
pbandk-runtime-jvm/0.10.0-SNAPSHOT/maven-metadata-local.xml
pbandk-runtime-jvm/0.10.0-SNAPSHOT/pbandk-runtime-jvm-0.10.0-SNAPSHOT-sources.jar
pbandk-runtime-jvm/0.10.0-SNAPSHOT/pbandk-runtime-jvm-0.10.0-SNAPSHOT.jar
pbandk-runtime-jvm/0.10.0-SNAPSHOT/pbandk-runtime-jvm-0.10.0-SNAPSHOT.module
pbandk-runtime-jvm/0.10.0-SNAPSHOT/pbandk-runtime-jvm-0.10.0-SNAPSHOT.pom
pbandk-runtime-jvm/maven-metadata-local.xml
pbandk-runtime-linuxx64
pbandk-runtime-linuxx64/0.10.0-SNAPSHOT
pbandk-runtime-linuxx64/0.10.0-SNAPSHOT/maven-metadata-local.xml
pbandk-runtime-linuxx64/0.10.0-SNAPSHOT/pbandk-runtime-linuxx64-0.10.0-SNAPSHOT-sources.jar
pbandk-runtime-linuxx64/0.10.0-SNAPSHOT/pbandk-runtime-linuxx64-0.10.0-SNAPSHOT.klib
pbandk-runtime-linuxx64/0.10.0-SNAPSHOT/pbandk-runtime-linuxx64-0.10.0-SNAPSHOT.module
pbandk-runtime-linuxx64/0.10.0-SNAPSHOT/pbandk-runtime-linuxx64-0.10.0-SNAPSHOT.pom
pbandk-runtime-linuxx64/maven-metadata-local.xml
pbandk-runtime-macosx64
pbandk-runtime-macosx64/0.10.0-SNAPSHOT
pbandk-runtime-macosx64/0.10.0-SNAPSHOT/maven-metadata-local.xml
pbandk-runtime-macosx64/0.10.0-SNAPSHOT/pbandk-runtime-macosx64-0.10.0-SNAPSHOT-sources.jar
pbandk-runtime-macosx64/0.10.0-SNAPSHOT/pbandk-runtime-macosx64-0.10.0-SNAPSHOT.klib
pbandk-runtime-macosx64/0.10.0-SNAPSHOT/pbandk-runtime-macosx64-0.10.0-SNAPSHOT.module
pbandk-runtime-macosx64/0.10.0-SNAPSHOT/pbandk-runtime-macosx64-0.10.0-SNAPSHOT.pom
pbandk-runtime-macosx64/maven-metadata-local.xml
pbandk-runtime-metadata
pbandk-runtime-metadata/0.10.0-SNAPSHOT
pbandk-runtime-metadata/0.10.0-SNAPSHOT/maven-metadata-local.xml
pbandk-runtime-metadata/0.10.0-SNAPSHOT/pbandk-runtime-metadata-0.10.0-SNAPSHOT-sources.jar
pbandk-runtime-metadata/0.10.0-SNAPSHOT/pbandk-runtime-metadata-0.10.0-SNAPSHOT.jar
pbandk-runtime-metadata/0.10.0-SNAPSHOT/pbandk-runtime-metadata-0.10.0-SNAPSHOT.module
pbandk-runtime-metadata/0.10.0-SNAPSHOT/pbandk-runtime-metadata-0.10.0-SNAPSHOT.pom
pbandk-runtime-metadata/maven-metadata-local.xml
protoc-gen-kotlin-jvm
protoc-gen-kotlin-jvm/0.10.0-SNAPSHOT
protoc-gen-kotlin-jvm/0.10.0-SNAPSHOT/maven-metadata-local.xml
protoc-gen-kotlin-jvm/0.10.0-SNAPSHOT/protoc-gen-kotlin-jvm-0.10.0-SNAPSHOT-jvm8.jar
protoc-gen-kotlin-jvm/0.10.0-SNAPSHOT/protoc-gen-kotlin-jvm-0.10.0-SNAPSHOT.pom
protoc-gen-kotlin-jvm/maven-metadata-local.xml
protoc-gen-kotlin-lib
protoc-gen-kotlin-lib/0.10.0-SNAPSHOT
protoc-gen-kotlin-lib/0.10.0-SNAPSHOT/maven-metadata-local.xml
protoc-gen-kotlin-lib/0.10.0-SNAPSHOT/protoc-gen-kotlin-lib-0.10.0-SNAPSHOT.jar
protoc-gen-kotlin-lib/0.10.0-SNAPSHOT/protoc-gen-kotlin-lib-0.10.0-SNAPSHOT.module
protoc-gen-kotlin-lib/0.10.0-SNAPSHOT/protoc-gen-kotlin-lib-0.10.0-SNAPSHOT.pom
protoc-gen-kotlin-lib/maven-metadata-local.xml
protoc-gen-kotlin-lib-jvm
protoc-gen-kotlin-lib-jvm/0.10.0-SNAPSHOT
protoc-gen-kotlin-lib-jvm/0.10.0-SNAPSHOT/maven-metadata-local.xml
protoc-gen-kotlin-lib-jvm/0.10.0-SNAPSHOT/protoc-gen-kotlin-lib-jvm-0.10.0-SNAPSHOT-sources.jar
protoc-gen-kotlin-lib-jvm/0.10.0-SNAPSHOT/protoc-gen-kotlin-lib-jvm-0.10.0-SNAPSHOT.jar
protoc-gen-kotlin-lib-jvm/0.10.0-SNAPSHOT/protoc-gen-kotlin-lib-jvm-0.10.0-SNAPSHOT.module
protoc-gen-kotlin-lib-jvm/0.10.0-SNAPSHOT/protoc-gen-kotlin-lib-jvm-0.10.0-SNAPSHOT.pom
protoc-gen-kotlin-lib-jvm/maven-metadata-local.xml
protoc-gen-kotlin-lib-metadata
protoc-gen-kotlin-lib-metadata/0.10.0-SNAPSHOT
protoc-gen-kotlin-lib-metadata/0.10.0-SNAPSHOT/maven-metadata-local.xml
protoc-gen-kotlin-lib-metadata/0.10.0-SNAPSHOT/protoc-gen-kotlin-lib-metadata-0.10.0-SNAPSHOT-sources.jar
protoc-gen-kotlin-lib-metadata/0.10.0-SNAPSHOT/protoc-gen-kotlin-lib-metadata-0.10.0-SNAPSHOT.jar
protoc-gen-kotlin-lib-metadata/0.10.0-SNAPSHOT/protoc-gen-kotlin-lib-metadata-0.10.0-SNAPSHOT.module
protoc-gen-kotlin-lib-metadata/0.10.0-SNAPSHOT/protoc-gen-kotlin-lib-metadata-0.10.0-SNAPSHOT.pom
protoc-gen-kotlin-lib-metadata/maven-metadata-local.xml
```

Reply:
I had checked the artifact name by printing them in the `publishing` block, not the actual resulting files.

I'll try to check again over the weekend.

Reply:
Since this would be just a minor cleanup, I'll go ahead and merge the rest of this PR so we can kick off some testing of the new version internally. If you don't mind, please still check the publishing thing when you have a chance and put up another PR if my suggestion works out.